service/auth.ts

```diff
@@ -13,6 +13,38 @@ import {
 } from "proto/auth_pb";
 import client from "service/client";
 
+interface CreateUserReq {
+  email: string;
+  username: string;
+  password: string;
+}
+
+interface CreateUserRes {
+  email: string;
+  username: string;
+  id: number;
+}
+
+export interface HttpError {
+  status_code: number;
+  errors: string[];
+  error_messages: {
+    [key: string]: string;
+  };
+}
```

Comment:
```suggestion
export interface HttpError {
  status_code: number;
  error_messages: string[];
  errors: {
    [key: string]: [string[]];
  };
}
```
I found in a backend PR where the standardised error processing is done, it states the error response should look like this instead.